gp-includes/warnings.php

```diff
@@ -401,8 +620,123 @@ function ( $key ) {
 			}
 		);
 
-		foreach ( $warnings as $warning ) {
-			$translation_warnings->add( str_replace( 'warning_', '', $warning ), array( $this, $warning ) );
+		$warnings = array_fill_keys( $warnings, $this );
+
+		$warnings = apply_filters( 'gp_add_all_warnings', $warnings );
+
+		foreach ( $warnings as $warning => $class ) {
+			$translation_warnings->add( str_replace( 'warning_', '', $warning ), array( $class, $warning ) );
 		}
 	}
+
+	/**
+	 * Returns the values from the href and the src
+	 *
+	 * @since 3.0.0
+	 * @access public
+	 *
+	 * @param array $content
+	 * @return array
+	 */
+	private function get_values_from_href_src( array $content ): array {
+		preg_match_all( '/<a[^>]+href=([\'"])(?<href>.+?)\1[^>]*>/i', implode( ' ', $content ), $href_values );
+		preg_match_all( '/<[^>]+src=([\'"])(?<src>.+?)\1[^>]*>/i', implode( ' ', $content ), $src_values );
+		return array_merge( $href_values['href'], $src_values['src'] );
+	}
+
+	/**
+	 * Checks if the HTML tags are in correct order
+	 *
+	 * Warns about HTML tags translations in incorrect order. For example:
+	 * - Original: <a></a>
+	 * - Translation: </a><a>
+	 *
+	 * @param array $original_parts     The original HTML tags
+	 * @param array $translation_parts  The translation HTML tags
+	 * @return string|true              True if check is OK, otherwise warning message.
+	 */
+	private function check_valid_html( array $original_parts, array $translation_parts ) {
+		if ( empty( $original_parts ) ) {
+			return true;
+		}
+		if ( $original_parts === $translation_parts ) {
+			return true;
+		}
+
+		libxml_clear_errors();
+		libxml_use_internal_errors( true );
+		$original = new DOMDocument();
+		$original->loadHTML( implode( '', $original_parts ) );
+		// If the original parts are not well-formed, don't continue the translation check
+		$errors = libxml_get_errors();
+		if ( ! empty( $errors ) ) {
+			return true;
+		}
+
+		$translation = new DOMDocument();
+		$translation->loadHTML( implode( '', $translation_parts ) );
+		$errors = libxml_get_errors();
+		if ( ! empty( $errors ) ) {
+			$message = array();
+			foreach ( $errors as $error ) {
+				$message[] = trim( $error->message );
+			}
+			return sprintf(
+				__( 'The translation tags are not correct: %s', 'glotpress' ),
```

Comment:
```suggestion
				__( 'The translation contains incorrect HTML tags: %s', 'glotpress' ),
```

This makes the message consistent with others and clarifies that this is about HTML tags .

Reply:
Done.